### ISSUE TITLE
Use the photo property when publishing a post with a single section

### ISSUE THREAD

**Issue by paulrobertlloyd** [microdotblog/sunlit#100]
Sunlit’s posting UI cleverly adapts when you add a second section, showing a field to enter a title. Conceptually, adding more than one section moves you into a mode where you are writing an [article post](https://indieweb.org/article), rather than a [photo post](https://indieweb.org/photo). However, the Micropub post request that Sunlit makes doesn’t reflect this difference.

## Properties sent when a post has one section

All information regarding the photo is combined into the `content` property. As the post contains only a `content` property (`name` is empty), it is deemed to be a [note post](https://indieweb.org/note).

Furthermore, the `html` provided doesn’t contain paragraph tags (#86), so while not invalid, it is certainly malformed.

```json
{
    "type": ["h-entry"],
    "properties": {
        "name": [""],
        "content": [{
            "html":"Tuneresque\n\n<img src=\"https://getindiekit.github.io/sandbox/media/photos/2020/08/30/8u53c.jpg\" width=\"600\" height=\"839\" alt=\"Dramatic sunset over Brighton’s West Pier\" style=\"height: auto;\" class=\"sunlit_image\" />"
        }],
        "post-status": ["published"]
    }
}
```

## Expected properties to be sent when a post has one section

I would expect the `photo` property to be provided, containing an array of photos, each with a URL to the image file, and its description, if provided. This can then be identified as a photo post, which is important should a server look to the post type to determine what permalink it is given, for example. This is what Micro.blog apps send.

```json
{
    "type": ["h-entry"],
    "properties": {
        "name": [""],
        "photo": [{
            "value": ["https://getindiekit.github.io/sandbox/media/photos/2020/08/30/8u53c.jpg"],
            "alt": ["Dramatic sunset over Brighton’s West Pier"],
        }],
        "content": ["Tuneresque"],
        "post-status": ["published"]
    }
}
```

## Posts with multiple sections

Once you add multiple sections, if you give a post a title, it is correctly identified as an article post (`name` + `content`). Should you omit the title, it is considered a note post (`content`). This behaviour seems correct.

**Comment by manton** [comment]
I left the `<p>` tags out because it's common to support Markdown when working with Micro.blog. I wonder if it should always include the tags when publishing to an external, non-Micro.blog endpoint.

As for the `photo` property… We could do that, but I don't think the current behavior is necessarily wrong either, since it makes it more consistent across different platforms.

**Comment by paulrobertlloyd** [comment]
Another advantage of using `photo` property is that authors publishing to compatible third-party websites can parse the value and present the images in their own way, without the need for class names, inline styles and set image widths.

Granted, I’ve not used a WordPress blog with the IndieWeb plugins though – not sure how they would deal with such information… HTML and inline images may be more predictable in that case.

**Comment by cheesemaker** [comment]
I have implemented this in the 3.2 beta. I'd like to see how well it works. If it's good, we can leave it, but if it's problematic, I'll likely just comment it out.

**Comment by paulrobertlloyd** [comment]
@cheesemaker Oh golly, this is *fantastic* news! I’m not able to use Sunlit at the moment because of the current behaviour, but this change would make Sunlit my go to app for posting photos. I could even see myself using it to create more complex posts, using the section titles and captions. 🎉
